js/components/common/react_modal.js.jsx

```diff
@@ -50,9 +50,16 @@ class CourseModal extends React.Component {
   }
 
   closeModal() {
+    this.props.callbackModal();
     this.setState({ modalIsOpen: false });
   }
 
+  componentDidUpdate (prevProps) {
+    if ((prevProps !== this.props) && (this.props.courseChange !== "")) {
```

Comment:
Never compare the prop object themselves, only specific values. In this case, you should compare the course code prop.